src/http_ece.rs

```diff
@@ -155,90 +108,44 @@ impl<'a> HttpEce<'a> {
-        context.push((self.peer_public_key.len() >> 8) as u8);
-        context.push((self.peer_public_key.len() & 0xff) as u8);
-        context.extend_from_slice(self.peer_public_key);
-        context.push((as_public_key.len() >> 8) as u8);
-        context.push((as_public_key.len() & 0xff) as u8);
-        context.extend_from_slice(as_public_key);
-
-        let client_auth_secret = hkdf::Salt::new(hkdf::HKDF_SHA256, &self.peer_secret);
-        let salt = hkdf::Salt::new(hkdf::HKDF_SHA256, salt_bytes);
-
-        let EceKey(prk) = client_auth_secret
-            .extract(shared_secret)
-            .expand(&[&"Content-Encoding: auth\0".as_bytes()], EceKey(32))
-            .unwrap()
-            .into();
-
-        let mut cek_info = Vec::with_capacity(165);
-        cek_info.extend_from_slice("Content-Encoding: aesgcm\0".as_bytes());
-        cek_info.extend_from_slice(&context);
-
-        let EceKey(content_encryption_key) = salt
-            .extract(&prk)
-            .expand(&[&cek_info], EceKey(16))
-            .unwrap()
-            .into();
-
-        let mut nonce_info = Vec::with_capacity(164);
-        nonce_info.extend_from_slice("Content-Encoding: nonce\0".as_bytes());
-        nonce_info.extend_from_slice(&context);
-
-        let EceKey(nonce_bytes) = salt
-            .extract(&prk)
-            .expand(&[&nonce_info], EceKey(12))
-            .unwrap()
-            .into();
-
-        let mut nonce = EceNonce::default();
-        nonce.fill(nonce_bytes);
-
-        let unbound_key = aead::UnboundKey::new(&aead::AES_128_GCM, &content_encryption_key)?;
-        let mut sealing_key = aead::SealingKey::new(unbound_key, nonce);
-
-        sealing_key.seal_in_place_append_tag(aead::Aad::empty(), payload)?;
-
-        Ok(())
-    }
-}
-
-fn front_pad(payload: &[u8], output: &mut [u8]) {
-    let payload_len = payload.len();
-    let max_payload = output.len() - 2;
-    let padding_size = max_payload - payload.len();
-
-    output[0] = (padding_size >> 8) as u8;
-    output[1] = (padding_size & 0xff) as u8;
-
-    for i in 0..payload_len {
-        output[padding_size + i + 2] = payload[i];
+    pub fn generate_headers_aes128gcm(&self) -> Vec<(&'static str, String)> {
+        let mut headers = Vec::new();
+        if let Some(signature) = &self.vapid_signature {
+            headers.push((
+                "Authorization",
+                format!("vapid t={}, k={}", signature.auth_t, signature.auth_k),
+            ));
+        }
+        headers
     }
 }
 
 #[cfg(test)]
 mod tests {
-    use base64::{self, URL_SAFE, URL_SAFE_NO_PAD};
+    use std::collections::HashMap;
+
     use crate::error::WebPushError;
-    use crate::http_ece::{front_pad, ContentEncoding, HttpEce};
+    use crate::http_ece::{ContentEncoding, HttpEce};
     use crate::vapid::VapidSignature;
+    use base64::{self, URL_SAFE};
+    use regex::Regex;
+
+    fn headers_to_hashmap(headers: Vec<(&'static str, String)>) -> HashMap<&'static str, String> {
```

Comment:
Do we actually need a hashmap?

Reply:
We don't need the function actually. I thought I removed it. 😅

Reply:
Oh no I see that's the one in the tests!  Well it's for tests simplicity, it's not included in the main code. I thought it was OK to keep it since the tests performance is probably not paramount. It's just to easily access the different header fields and test them in the headers tests without having to search them in a vec

Reply:
Since introducing an order in fields does not seem like a good idea for unordered data and it might change in the future, the alternative is to use a search function and run it for each field but we're probably loosing readability... Personally having a Hashmap in a test mod does not bother me too much 😅 tell me what you think ;)

Reply:
Nah it's only tests. That's fine.